cylc/flow/scripts/lint.py

```diff
@@ -0,0 +1,392 @@
+            'short': '``[cylc]simulation`` is obsolete.',
+            'url': ''
+        },
+        re.compile(r'spawn to max active cycle points\s*?='): {
+            'short': '``[cylc]spawn to max active cycle points`` is obsolete.',
+            'url': (
+                'https://cylc.github.io/cylc-doc/latest/html/reference'
+                '/config/workflow.html#flow.cylc[scheduling]runahead%20limit'
+            ),
+        },
+        re.compile(r'abort on stalled\s*?='): {
+            'short':
+                '``[cylc][events]abort on stalled`` is obsolete.',
+            'url': ''
+        },
+        re.compile(r'abort if .* handler fails\s*?='): {
+            'short': (
+                '``[cylc][events]abort on ___ handler fails`` commands are'
+                ' obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'.* handler\s*?='): {
+            'short': (
+                '``[cylc][<namespace>][events]___ handler`` commands are'
+                ' now "handlers".'
+            ),
+            'url': ''
+        },
+        re.compile(r'mail retry delays\s*?='): {
+            'short': (
+                '``[runtime][<namespace>][events]mail retry delays`` '
+                'is obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'extra log files\s*?='): {
+            'short': (
+                '``[runtime][<namespace>][events]extra log files`` '
+                'is obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'shell\s*?='): {
+            'short': (
+                '``[runtime][<namespace>]shell`` '
+                'is obsolete.'
+            ),
+            'url': ''
+        },
+        re.compile(r'suite definition directory\s*?='): {
+            'short': (
+                '``[runtime][<namespace>][remote]suite definition directory`` '
+                'is obsolete.'
+            ),
+            'url': 'summary.html#symlink-dirs'
+        },
+        re.compile(SECTION2.format('dependencies')): {
+            'short': '``[dependencies]`` is deprecated.',
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(r'graph\s*?='): {
+            'short': (
+                '``[cycle point]graph =`` is deprecated, '
+                'use ``cycle point = <graph>``'
+            ),
+            'url': 'major-changes/config-changes.html#graph'
+        },
+        re.compile(SECTION2.format('remote')): {
+            'short': (
+                '``[runtime][<namespace>][remote]host`` is deprecated, '
+                'use ``[runtime][<namespace>]platform``'
+            ),
+            'url': 'major-changes/platforms.html#platforms'
+        },
+        re.compile(SECTION3.format('job')): {
+            'short': (
+                '``[runtime][<namespace>][job]`` is deprecated, '
+                'use ``[runtime][<namespace>]platform``'
```

Comment:
What about https://github.com/cylc/cylc-flow/blob/0c977d6956e3d714946dcf16526b114ba1055537/tests/functional/deprecations/01-cylc8-basic/validation.stderr#L28-L32

Reply:
In general this is correct, former job settings should be covered by platforms. In the case of `execution time limit` there should be a message to move it up one level. In the case of `submission time limit` the message should be something like  

> In most cases you can use platform default `submission time limit`s, but you can set it yourself using 
> `[task/FAMILY]submission timeout limit."